src/IGCompiler.ts

```
@@ -0,0 +1,168 @@
+ *
+ */
+export class IGCompiler {
+    private options: IGCompilerOptions;
+
+    private readonly implementationGuides: ImplementationGuides;
+
+    constructor(implementationGuides: ImplementationGuides, options: IGCompilerOptions) {
+        this.options = options;
+        this.implementationGuides = implementationGuides;
+    }
+
+    async collectResources(igDir: PathLike, resources: any[]): Promise<void> {
+        const indexJson = path.join(igDir.toString(), '.index.json');
+        if (!existsSync(indexJson)) {
+            return;
+        }
+        const index: any = await loadJson(indexJson);
+        for (const file of index.files) {
+            if (file.resourceType === 'SearchParameter') {
+                const filePath = path.join(igDir.toString(), file.filename);
+                console.log(`Compiling ${filePath}`);
+                resources.push(await loadJson(filePath));
+            }
+        }
+    }
+
+    /**
+     * Main public function
+     * @param igsDir
+     * @param outputPath
+     */
+    public async compileIGs(igsDir: PathLike, outputPath: PathLike): Promise<void> {
+        const resources: any[] = [];
+        if (!existsSync(igsDir)) {
+            throw new Error(`'${igsDir}' doesn't exist`);
+        }
+        for (const igInfo of await this.collectIGInfos(igsDir)) {
+            await this.collectResources(igInfo.path, resources);
+        }
+        const compiledResources = await this.implementationGuides.compile(resources);
+        await storeJson(outputPath, compiledResources);
+    }
+
+    createIGKey(name: string, version: string) {
+        if (this.options.ignoreVersion) {
+            return name;
+        }
+        return `${name}@${version}`;
+    }
+
+    async extractIgInfo(igDir: PathLike): Promise<IGInfo | null> {
+        const packagePath = path.join(igDir.toString(), 'package.json');
+        if (!existsSync(packagePath)) {
+            return null;
```

We should throw an error here. Same reason as my other comment regarding a missing `index.json` file